### PR TITLE
LUT-27586: Updated the form used in the task to send a custom (manual) email

### PR DESCRIPTION
**Pull request by atos-mohamed-b**
The following PR should be merged first:
- https://github.com/lutece-secteur-public/gru-module-workflow-appointment/pull/12

### CODE REVIEW

File: src/java/fr/paris/lutece/plugins/workflow/modules/appointment/provider/AppointmentNotificationMarkers.java [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+        collectionNotifyMarkers.add( createMarkerValues( AppointmentWorkflowConstants.MARK_LASTNAME, _appointment.getLastName( ) ) );
+        collectionNotifyMarkers.add( createMarkerValues( AppointmentWorkflowConstants.MARK_EMAIL, _appointment.getEmail( ) ) );
+        collectionNotifyMarkers.add( createMarkerValues( AppointmentWorkflowConstants.MARK_DATE_APPOINTMENT, _appointment.getDateOfTheAppointment( ) ) );
+        collectionNotifyMarkers.add( createMarkerValues( AppointmentWorkflowConstants.MARK_TIME_APPOINTMENT, _appointment.getStartingTime( ).toString( ) ) );
+        collectionNotifyMarkers.add( createMarkerValues( AppointmentWorkflowConstants.MARK_END_TIME_APPOINTMENT, _appointment.getEndingTime( ).toString( ) ) );
+        collectionNotifyMarkers.add( createMarkerValues( AppointmentWorkflowConstants.MARK_REFERENCE, _appointment.getReference( ) ) );
+        // Retrieve the URLs to cancel, report... an appointment
+        String strUrlReport = AppointmentApp.getReportAppointmentUrl( _appointment );
+        collectionNotifyMarkers.add( createMarkerValues( AppointmentWorkflowConstants.MARK_URL_REPORT, strUrlReport.replaceAll( "&", "&amp;" ) ) );
```
Comment (seboo) [inline src/java/fr/paris/lutece/plugins/workflow/modules/appointment/provider/AppointmentNotificationMarkers.java:124]: Encoding the "&" char shoud be done in the template, like "${url?html}"
(the url value can be used in <href> tag)

File: src/java/fr/paris/lutece/plugins/workflow/modules/appointment/provider/AppointmentNotificationMarkers.java [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+     * @param strMarkerName
+     *            The marker's name
+     * @param strMarkerValue
+     *            The marker's value
+     * @return the InfoMarker Object
+     */
+    private static InfoMarker createMarkerValues( String strMarkerName, String strMarkerValue )
+    {
+        InfoMarker infoMarker = new InfoMarker( strMarkerName );
+        infoMarker.setValue( strMarkerValue );
```
Comment (seboo) [inline src/java/fr/paris/lutece/plugins/workflow/modules/appointment/provider/AppointmentNotificationMarkers.java:161]: Use new setData( ) method to send the object in the model to the template

File: src/java/fr/paris/lutece/plugins/workflow/modules/appointment/provider/AppointmentNotificationMarkers.java [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+     */
+    public Collection<InfoMarker> getMarkerValues( )
+    {
+        Collection<InfoMarker> collectionNotifyMarkers = new ArrayList<>( );
+
+        collectionNotifyMarkers.add( createMarkerValues( AppointmentWorkflowConstants.MARK_FIRSTNAME, _appointment.getFirstName( ) ) );
+        collectionNotifyMarkers.add( createMarkerValues( AppointmentWorkflowConstants.MARK_LASTNAME, _appointment.getLastName( ) ) );
+        collectionNotifyMarkers.add( createMarkerValues( AppointmentWorkflowConstants.MARK_EMAIL, _appointment.getEmail( ) ) );
+        collectionNotifyMarkers.add( createMarkerValues( AppointmentWorkflowConstants.MARK_DATE_APPOINTMENT, _appointment.getDateOfTheAppointment( ) ) );
+        collectionNotifyMarkers.add( createMarkerValues( AppointmentWorkflowConstants.MARK_TIME_APPOINTMENT, _appointment.getStartingTime( ).toString( ) ) );
```
Comment (seboo) [inline src/java/fr/paris/lutece/plugins/workflow/modules/appointment/provider/AppointmentNotificationMarkers.java:119]: send the date as object

File: src/java/fr/paris/lutece/plugins/workflow/modules/appointment/provider/AppointmentNotificationMarkers.java [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+        collectionNotifyMarkers.add( createMarkerValues( AppointmentWorkflowConstants.MARK_END_TIME_APPOINTMENT, _appointment.getEndingTime( ).toString( ) ) );
```
Comment (seboo) [inline src/java/fr/paris/lutece/plugins/workflow/modules/appointment/provider/AppointmentNotificationMarkers.java:120]: idem

File: src/java/fr/paris/lutece/plugins/workflow/modules/appointment/provider/AppointmentNotificationMarkers.java [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+     * @param strMarkerName
+     *            The marker's name
+     * @param strMarkerDescription
+     *            The marker's description
+     * @return the InfoMarker Object
+     */
+    private static InfoMarker createMarkerDescriptions( String strMarkerName, String strMarkerDescription )
+    {
+        InfoMarker infoMarker = new InfoMarker( strMarkerName );
+        infoMarker.setDescription( I18nService.getLocalizedString( strMarkerDescription, I18nService.getDefaultLocale( ) ) );
```
Comment (seboo) [inline src/java/fr/paris/lutece/plugins/workflow/modules/appointment/provider/AppointmentNotificationMarkers.java:144]: the locale should be sent as parameter

**Review by seboo — CHANGES_REQUESTED** [review]

**Comment by atos-mohamed-b** [comment]
The PR has now been rebased on the previous code:
- https://github.com/lutece-secteur-public/gru-module-workflow-appointment/pull/12

A Redmine issue to rework the module has been created:
- https://dev.lutece.paris.fr/bugtracker/issues/27872

**Pull request merged**